mongo_connector/dest_mapping.py

```diff
@@ -0,0 +1,186 @@
+import logging
+import threading
+import re
+from mongo_connector import errors
+
+
+LOG = logging.getLogger(__name__)
+
+"""New structure to handle both plain and wildcard mapping namespaces dynamically.
+"""
+
+class DestMapping():
+    def __init__(self, namespace_set=[], ex_namespace_set=[], inpt={}):
+        # a dict containing plain mappings
+        self.plain = {}
+        # a dict containing wildcard mappings
+        self.wildcard = {}
+        # a dict containing reverse plain mapping
+        # because the mappings are not duplicated, so the values should also be unique
+        self.reverse_plain = {}
+        # a dict containing plain db mappings, db -> an array of mapped db
+        self.plain_db = {}
+        
+        # the input namespace_set and ex_namespace_set could contain wildcard
+        self.namespace_set = namespace_set
+        self.ex_namespace_set = ex_namespace_set
+        
+        self.lock = threading.Lock()
+        
+        # initialize
+        for ns in namespace_set:
+            if ns not in inpt.keys():
+                inpt[ns] = ns
+
+        for k, v in inpt.items():
+            self.set(k,v)
+            
+    def match(self, src, dst_pattern):
+        """If source string src matches dst, return the matchobject"""
+        d_arr = re.split('(\*)',dst_pattern)
+        r_arr = []
+        for x in d_arr:
+            if x != "*":
+                r_arr.append(re.escape(x))
+            else:
+                r_arr.append("([^*]+)")
```

Comment:
I think this can be easier. The patterns `\A` and `\Z` can be used to represent the beginning/end of a string in Python, respectively. So, I think `reg_pattern` could just be `r'\A' + dst_pattern.replace('*', '.*') + r'\Z'`.